storage/db-tool/src/tests.rs

```diff
@@ -60,3 +74,75 @@ fn test_various_cmd_parsing() {
 fn run_cmd(args: &[&str]) {
     DBTool::try_parse_from(args).expect("command parse unsuccessful");
 }
+
+#[tokio::test]
+async fn test_backup_compaction() {
+    let db = test_execution_with_storage_impl();
+    let backup_dir = TempPath::new();
+    backup_dir.create_as_dir().unwrap();
+    let local_fs = LocalFs::new(backup_dir.path().to_path_buf());
+    let store: Arc<dyn BackupStorage> = Arc::new(local_fs);
+    let rt = start_backup_service(SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 6186), db);
+
+    // Backup the local_test DB
+    DBTool::try_parse_from([
+        "aptos-db-tool",
+        "backup",
+        "oneoff",
+        "epoch-ending",
+        "--start-epoch",
+        "0",
+        "--end-epoch",
+        "2",
+        "--local-fs-dir",
+        backup_dir.path().to_str().unwrap(),
+    ])
+    .unwrap()
+    .run()
+    .await
+    .unwrap();
+    DBTool::try_parse_from([
+        "aptos-db-tool",
+        "backup",
+        "oneoff",
+        "state-snapshot",
+        "--state-snapshot-epoch",
+        "2",
+        "--local-fs-dir",
+        backup_dir.path().to_str().unwrap(),
+    ])
+    .unwrap()
+    .run()
+    .await
+    .unwrap();
+    DBTool::try_parse_from([
+        "aptos-db-tool",
+        "backup",
+        "oneoff",
+        "transaction",
+        "--start-version",
+        "0",
+        "--num_transactions",
+        "30",
+        "--local-fs-dir",
+        backup_dir.path().to_str().unwrap(),
+    ])
+    .unwrap()
+    .run()
+    .await
+    .unwrap();
+    // assert the metadata views are same before and after compaction
+    let metadata_opt = MetadataCacheOpt::new(Some(TempPath::new().path().to_path_buf()));
+    let old_metaview = metadata::cache::sync_and_load(&metadata_opt, Arc::clone(&store), 1)
+        .await
+        .unwrap();
+    let compactor = BackupCompactor::new(2, 2, 2, metadata_opt.clone(), Arc::clone(&store), 1);
+    compactor.run().await.unwrap();
```

Comment:
hmm.. is there actually anything to compact here? Did I misread it?

Reply:
add more backup to make the compaction more interesting...